openassessment/fileupload/api.py

```diff
@@ -29,3 +37,299 @@ def remove_file(key):
     Remove file from the storage
     """
     return backends.get_backend().remove_file(key)
+
+
+def get_student_file_key(student_item_dict, index=0):
+    """
+    Args:
+        student_item_dict: A dictionary containing keys ('student_id', 'course_id', 'item_id').
+        index (int, optional): The index of a file.
+    """
+    key_template = '{student_id}/{course_id}/{item_id}'
+    index = int(index)
+    if index > 0:
+        key_template += '/{index}'
+    return key_template.format(index=index, **student_item_dict)
+
+
+def _safe_load_json_list(field):
+    """
+    Tries to load JSON-ified string,
+    returns an empty list if we try to load some non-JSON-encoded string.
+    """
+    try:
+        return json.loads(field)
+    except ValueError:
+        return []
+
+
+class FileUpload(object):
+    """
+    A layer of abstraction over the various components of file
+    data stored as ORA XBlock user-scoped fields.
+    """
+    def __init__(self, name=None, description=None, size=None, index=None, descriptionless=False, **student_item_dict):
```

Comment:
Is the "descriptionless" parameter there to handle the boundary condition that we fixed? I am trying to think if there is a slightly better / descriptive (pardon the pun) name ?

Reply:
Yes it is.  I couldn't think of a better one, but lemme know if you have suggestions.